x11-packages/xfce4-cpugraph-plugin/0001-cpu-data-thread-stat.patch

```diff
@@ -0,0 +1,181 @@
+diff --git a/panel-plugin/os.cc b/panel-plugin/os.cc
+index a8c9e40..3016be9 100644
+--- a/panel-plugin/os.cc
++++ b/panel-plugin/os.cc
+@@ -79,9 +79,176 @@ static kstat_ctl_t *kc;
+ #endif
+ 
+ #if defined (__linux__) || defined (__FreeBSD_kernel__)
++
++#ifdef __ANDROID__
++
++#include <sys/system_properties.h>
++#include <unistd.h>
++#include <glob.h>
++
++static int _sdkver = -1;
++
++void get_sdk_ver() {
```

Comment:
`void get_...` probably isn't right. `set_sdk_ver()` if you wanna keep it void, but given how you're using it:

```
int get_sdk_ver() {
  static int sdk_ver = -1;
  if (sdk_ver == -1) {
    // do the thing
  }
  return sdk_ver;
}
```

seems better.

Reply:
Such style is copied from `xfce4-taskamanger` casually without much thought. Agree with you and will change it later